controllers/opentelemetrycollector_controller_test.go

```diff
@@ -229,60 +251,130 @@ func TestNewStatefulSetObjectsOnReconciliation(t *testing.T) {
 		assert.NotEmpty(t, list.Items)
 	}
 	{
-		list := &appsv1.StatefulSetList{}
+		list := &appsv1.DeploymentList{}
 		err = k8sClient.List(context.Background(), list, opts...)
 		assert.NoError(t, err)
 		assert.NotEmpty(t, list.Items)
 	}
 	{
-		list := &appsv1.DeploymentList{}
+		list := &appsv1.DaemonSetList{}
 		err = k8sClient.List(context.Background(), list, opts...)
 		assert.NoError(t, err)
-		// attention! we expect deployments to be empty when starting in the statefulset mode.
+		// attention! we expect daemonsets to be empty in the default configuration
 		assert.Empty(t, list.Items)
 	}
 	{
-		list := &appsv1.DaemonSetList{}
+		list := &appsv1.StatefulSetList{}
 		err = k8sClient.List(context.Background(), list, opts...)
 		assert.NoError(t, err)
-		// attention! we expect daemonsets to be empty when starting in the statefulset mode.
+		// attention! we expect statefulsets to be empty in the default configuration
+		assert.Empty(t, list.Items)
+	}
+	{
+		list := &routev1.RouteList{}
+		err = k8sClient.List(context.Background(), list, opts...)
+		// routes should be empty in this test because we explicitly do not have openshift routes available.
+		assert.NoError(t, err)
 		assert.Empty(t, list.Items)
 	}
 
 	// cleanup
 	require.NoError(t, k8sClient.Delete(context.Background(), created))
 
+	// cleanup the deployment deliberately, otherwise a local tester will always fail as there is no gc event.
+	list := &appsv1.DeploymentList{}
+	err = k8sClient.List(context.Background(), list, opts...)
+	assert.NoError(t, err)
+	assert.Len(t, list.Items, 1)
+	require.NoError(t, k8sClient.Delete(context.Background(), list.Items[0].DeepCopy()))
 }
 
-func TestContinueOnRecoverableFailure(t *testing.T) {
+func TestNewStatefulSetObjectsOnReconciliation(t *testing.T) {
```

Comment:
This is a bad diff from GitHub, I only added a test but it's not recognizing it as an add